tiledb/sm/misc/types.h

```diff
@@ -318,6 +320,21 @@ class ByteVecValue {
     x_.clear();
   }
 
+  // To string function
+  std::string to_hex_str() const {
+    std::stringstream ss;
+    for (size_t i = 0; i < x_.size(); i++) {
+      if (x_[i] < 16) {
+        ss << "0";
+      }
+      ss << std::hex << +x_[i];
+      if (i != x_.size() - 1) {
+        ss << " ";
+      }
+    }
+    return ss.str();
+  }
+
```

Comment:
Is `to_hex_str` only used for testing?  Maybe it should be a free function in the test file instead?  (iostream and sstream includes could be removed as well.)

Reply:
Yes, it does seem to be only used for testing.

Reply:
There is a problem with doing this--mainly, the issue is that implementing the query ast node print functions gets more difficult because of the multiple inheritance.

Reply:
Nevermind this has been done